src/pages/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from 'react';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import {
+  TextField,
+  IconButton,
+  List,
+  ListItem,
+  ListItemText,
+} from '@material-ui/core';
+import AddIcon from '@material-ui/icons/Add';
+import DeleteIcon from '@material-ui/icons/Delete';
+
+interface TodoItem {
+  title: string;
+  id: number;
+  done: boolean;
+}
+
+interface MyFormValues {
+  title: string;
+}
+
+const TodoList: React.FC = () => {
+  const [list, setList] = useState<TodoItem[]>([]);
+  const [itemFilter, setItemFilter] = useState('');
+  const initialValues: MyFormValues = { title: '' };
+  const formik = useFormik({
+    initialValues,
+    onSubmit: (values, actions) => {
+      const newTodo: TodoItem = {
+        title: values.title,
+        id: Math.random(),
+        done: false,
+      };

```

Comment:
I suggest using `uuid` package and organizing props:

```suggestion
      const newTodo: TodoItem = {
        id: uuidv4(),
        title: values.title,
        done: false,
      };

```